### PR TITLE
encode() and a stub for the load()

### PR DESCRIPTION
**Pull request by bzz**
Another part of #83

 - `Context.encode()` implementation  + tests
 - a stub for `Node.load()`

Actual load() implementation will be done in subsequent PRs.

### CODE REVIEW

File: src/main/scala/org/bblfsh/client/v2/JNode.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.bblfsh.client.v2
+
+sealed abstract class JNode
```
Comment (creachadair) [inline src/main/scala/org/bblfsh/client/v2/JNode.scala]: Can you please add a comment to say what this is for, even if only briefly?

File: src/main/scala/org/bblfsh/client/v2/package.scala [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package org.bblfsh.client
+
+package object v2 {
```
Comment (creachadair) [inline src/main/scala/org/bblfsh/client/v2/package.scala]: Can you please comment what this is for?

File: src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc [hunk 3]
```diff
@@ -156,6 +149,28 @@ JNIEXPORT jobject JNICALL Java_org_bblfsh_client_v2_Context_root(JNIEnv *env,
   return p->RootNode();
 }
 
+JNIEXPORT jobject JNICALL Java_org_bblfsh_client_v2_Context_encode(
+    JNIEnv *env, jobject self, jobject node) {
+  UastFormat fmt = UAST_BINARY;  // TODO(bzz): make it argument & enum
+
+  ContextExt *p = getHandle<ContextExt>(env, self, "nativeContext");
+  return p->Encode(node, fmt);
+  return nullptr;
```
Comment (dennwc) [inline src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc]:
```suggestion
```

File: src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc [hunk 4]
```diff
@@ -156,6 +149,28 @@ JNIEXPORT jobject JNICALL Java_org_bblfsh_client_v2_Context_root(JNIEnv *env,
...
+}
+
+JNIEXPORT void JNICALL Java_org_bblfsh_client_v2_Context_dispose(JNIEnv *env,
+                                                                 jobject self) {
+  ContextExt *p = getHandle<ContextExt>(env, self, "nativeContext");
+  setHandle<ContextExt>(env, self, 0, "nativeContext");
```
Comment (dennwc) [inline src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc]: Maybe move `"nativeContext"` to a constant?

File: src/main/scala/org/bblfsh/client/v2/package.scala (hunk 2 above)
Comment (bzz) [reply src/main/scala/org/bblfsh/client/v2/package.scala]: This only serves for defining package-wide type aliases - do you think it is still worth documenting that, even if that is a standard way of doing so in Scala?

Will add a doc for the type alias.

Comment (bzz) [reply src/main/scala/org/bblfsh/client/v2/package.scala]: addressed in 8162713fa88a425510391838a9f51c9bb83c5466

File: src/main/scala/org/bblfsh/client/v2/JNode.scala (hunk 1 above)
Comment (bzz) [reply src/main/scala/org/bblfsh/client/v2/JNode.scala]: addressed in 8162713fa88a425510391838a9f51c9bb83c5466

File: src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc (hunk 4 above)
Comment (bzz) [reply src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc]: addressed in 0d76c98dbdf5ec8db5dedb43dda35a26d5f59c13

**Comment by bzz** [comment]
Thank you for prompt reviews! 
@dennwc all feedback addressed, ready for another round.

File: src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc [hunk 5]
```diff
@@ -15,6 +15,8 @@ jobject asJvmBuffer(uast::Buffer buf) {
   return env->NewDirectByteBuffer(buf.ptr, buf.size);
 }
 
+const char *nativeContext = "nativeContext";
```
Comment (creachadair) [inline src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc]: Minor nit: Either `const char * const nativeContext = "…"` or `const char nativeContext[] = "…"` is preferable here—to ensure the pointer itself is also const (either works). `constexpr char nativeContext[] = "…"` will also work if we assume C++ ≥ 11.

File: src/main/scala/org/bblfsh/client/v2/package.scala (hunk 2 above)
Comment (creachadair) [reply src/main/scala/org/bblfsh/client/v2/package.scala]: If this is conventional (and would thus not surprise someone actually familiar with Scala, unlike me) then I don't think there's any need to document it specially.

File: src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc (hunk 5 above)
Comment (bzz) [reply src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc]: Yes, thank you, we do! Was thinking about `constexpr` as well - will replace as part of the next PR.

**Pull request merged**